src/components/transport_manager/src/websocket/websocket_listener.cc

```diff
@@ -0,0 +1,300 @@
+
+void WebSocketListener::Terminate() {
+  LOG4CXX_AUTO_TRACE(logger_);
+  Shutdown();
+}
+
+TransportAdapter::Error WebSocketListener::StartListening() {
+  LOG4CXX_AUTO_TRACE(logger_);
+  if (acceptor_.is_open()) {
+    return TransportAdapter::OK;
+  }
+
+#ifdef ENABLE_SECURITY
+  auto const ta_error = AddCertificateAuthority();
+  if (TransportAdapter::OK != ta_error) {
+    return ta_error;
+  }
+#endif
+
+  auto const address =
+      boost::asio::ip::make_address(settings_.websocket_server_address());
+  tcp::endpoint endpoint = {address, settings_.websocket_server_port()};
+
+  // Open the acceptor
+  boost::system::error_code ec;
+  acceptor_.open(endpoint.protocol(), ec);
+  if (ec) {
+    auto str_err = "ErrorOpen: " + ec.message();
+    LOG4CXX_ERROR(logger_,
+                  str_err << " host/port: " << endpoint.address().to_string()
+                          << "/" << endpoint.port());
+    return TransportAdapter::FAIL;
+  }
+
+  acceptor_.set_option(boost::asio::socket_base::reuse_address(true), ec);
+  if (ec) {
+    std::string str_err = "ErrorSetOption: " + ec.message();
+    LOG4CXX_ERROR(logger_,
+                  str_err << " host/port: " << endpoint.address().to_string()
+                          << "/" << endpoint.port());
+    return TransportAdapter::FAIL;
+  }
+
+  // Bind to the server address
+  acceptor_.bind(endpoint, ec);
+  if (ec) {
+    std::string str_err = "ErrorBind: " + ec.message();
+    LOG4CXX_ERROR(logger_,
+                  str_err << " host/port: " << endpoint.address().to_string()
+                          << "/" << endpoint.port());
+    return TransportAdapter::FAIL;
+  }
+
+  // Start listening for connections
+  acceptor_.listen(boost::asio::socket_base::max_listen_connections, ec);
+  if (ec) {
+    std::string str_err = "ErrorListen: " + ec.message();
+    LOG4CXX_ERROR(logger_,
+                  str_err << " host/port: " << endpoint.address().to_string()
+                          << "/" << endpoint.port());
+    return TransportAdapter::FAIL;
+  }
+
+  if (false == Run()) {
+    return TransportAdapter::FAIL;
+  }
+
+  return TransportAdapter::OK;
+}
+
+#ifdef ENABLE_SECURITY
+TransportAdapter::Error WebSocketListener::AddCertificateAuthority() {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  const auto cert_path = settings_.ws_server_cert_path();
+  LOG4CXX_DEBUG(logger_, "Path to certificate : " << cert_path);
+  const auto key_path = settings_.ws_server_key_path();
+  LOG4CXX_DEBUG(logger_, "Path to key : " << key_path);
+  const auto ca_cert_path = settings_.ws_server_ca_cert_path();
+  LOG4CXX_DEBUG(logger_, "Path to ca cert : " << ca_cert_path);
+  start_secure_ =
+      !cert_path.empty() && !key_path.empty() && !ca_cert_path.empty();
+
+  if (start_secure_ && (!file_system::FileExists(cert_path) ||
+                        !file_system::FileExists(key_path) ||
+                        !file_system::FileExists(ca_cert_path))) {
+    LOG4CXX_ERROR(logger_, "Certificate or key not found");
```

Comment:
add certificate or key file

Reply:
Fixed in 32836e4d0626a660bbc5e7ae2a976dac1acb79ad